overlord/auth/auth_test.go

```diff
@@ -455,3 +455,21 @@ func (as *authSuite) TestEnsureContexts(c *C) {
 
 	c.Check(auth.IsEnsureContext(context.TODO()), Equals, false)
 }
+
+func (as *authSuite) TestIdOnly(c *C) {
+	input := &auth.UserState{
+		ID:              1,
+		Username:        "username",
+		Email:           "email@test.com",
+		Macaroon:        "1234",
+		Discharges:      nil,
+		StoreMacaroon:   "macaroon",
+		StoreDischarges: []string{"discharge"},
+	}
+	expected := &auth.UserState{
+		ID:       1,
+		Username: "username",
+		Email:    "email@test.com",
+	}
+	c.Check(input.IdOnly(), DeepEquals, expected)
```

Comment:
remove this test please, the behavior was tested indirectly through the Remove* method tests, there's no reason to explicitly test this internal detail

Reply:
Isn't that what unit testing is all about? Internal details are tested all the time.

Reply:
As I said, the code was essentially tested anyway but not explicitly. We don't test all our helpers explicitly, there are various philosophies that can be applied, some generate more obvious coverage at the cost of more test changes on refactoring. It's a judgement call but overall our approach of almost always trying to write tests in foo_test and not foo means we are trying be conscious whether internal details merit a test or not.

Reply:
removed